applications/nrf5340_audio/src/bluetooth/le_audio_cis_headset.c

```diff
@@ -455,16 +454,41 @@ static void stream_recv_cb(struct bt_audio_stream *stream, const struct bt_iso_r
 	}
 }
 
-static void stream_start_cb(struct bt_audio_stream *stream)
+static void stream_start_cb(struct bt_bap_stream *stream)
 {
+	int ret;
+	
 	LOG_INF("Stream %p started", stream);
+
+	ret = ctrl_events_le_audio_event_send(LE_AUDIO_EVT_STREAMING);
+	ERR_CHK(ret);
+}
+
+static void stream_released_cb(struct bt_bap_stream *stream)
+{
+	LOG_INF("Stream %p released", stream);
+}
+
+static void stream_enabled_cb(struct bt_bap_stream *stream)
+{
+	int ret;
+	LOG_DBG("Stream %p enabled", stream);
+
+	if (stream->ep->dir == BT_AUDIO_DIR_SINK) {
+		/* Automatically do the receiver start ready operation */
+		ret = bt_bap_stream_start(stream);
+		if (ret != 0) {
+			LOG_ERR("Failed to start stream: %d", ret);
+			return;
+		}
+	}
 }
 
-static void stream_stop_cb(struct bt_audio_stream *stream)
+static void stream_stop_cb(struct bt_bap_stream *stream, uint8_t reason)
 {
 	int ret;
 
-	LOG_DBG("Stream %p stopping", stream);
+	LOG_INF("Stream stopped. Reason: %d", reason);
```

Comment:
I think this should be DBG. The INF should be in streamctrl, when LE_AUDIO_EVT_NOT_STREAMING is received.

Reply:
Alternatively, the reason could be DBG

Reply:
I also see that we have callbacks for changes to the stream, but also callbacks for changes to the unicast_server. I think we need to go through and decide which one of them to use, as I think they are both called most of the times. Maybe @Thalley can provide more information.

Reply:
As discussed on Teams, this will be changed when we do the architecture rewrite